### PR TITLE
refactor(ui): add new ChannelSummaryList component

### PR DESCRIPTION
**Pull request by mrfelton**
## Description:

Add a new ChannelSummaryList component that can displays a list of channels in a summary view.

## Motivation and Context:

Needed for new Channel UI - Re #1425

## How Has This Been Tested?

Storybook

## Screenshots (if appropriate):

image

## Types of changes:

New components

## Checklist:

- [x] My code follows the code style of this project.
- [x] I have reviewed and updated the documentation accordingly.
- [x] I have read the _CONTRIBUTING_ document.
- [ ] I have added tests to cover my changes where needed.
- [x] All new and existing tests passed.
- [x] My commits have been squashed into a concise set of changes.

### CODE REVIEW

File: app/components/Channels/ChannelSummaryList.js [hunk 1]
```diff
@@ -0,0 +1,54 @@
...
+}) => (
+  <Box as="article" {...rest}>
+    {channels.map((channelObj, index) => {
+      const channel = Object.prototype.hasOwnProperty.call(channelObj, 'channel')
+        ? channelObj.channel
+        : channelObj
+
+      return (
+        <ChannelSummaryListItem
+          key={index}
```
Comment (korhaliv) [inline app/components/Channels/ChannelSummaryList.js]: Can we use something other than index? `chan_id` looks good for this purpose. Using indexes as a key is generally an antipattern

File: app/components/Channels/ChannelSummaryList.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+  channels,
+  cryptoCurrency,
+  cryptoCurrencies,
+  setCryptoCurrency,
+  showChannelDetail,
+  ...rest
+}) => (
+  <Box as="article" {...rest}>
+    {channels.map((channelObj, index) => {
+      const channel = Object.prototype.hasOwnProperty.call(channelObj, 'channel')
```
Comment (korhaliv) [inline app/components/Channels/ChannelSummaryList.js]: Is there a need for `hasOwnProperty` vs `in` here ? If we indeed don't want to take into account proto property then maybe it's worth to add a comment why

Comment (mrfelton) [reply app/components/Channels/ChannelSummaryList.js]: Have simplified this.

File: app/components/Channels/ChannelSummaryList.js (hunk 1 above)
Comment (mrfelton) [reply app/components/Channels/ChannelSummaryList.js]: `chan_id` isn't always set (not set for pending nodes for example). I've created a new key which is a combination of a few different things which I think should be guaranteed to be unique.

**Pull request merged**